packages/devtools_app/lib/src/shared/table/table_controller.dart

```diff
@@ -72,7 +90,16 @@ class FlatTableController<T> extends TableControllerBase<T> {
     ColumnData<T>? secondarySortColumn,
     String? dataKey,
   }) {
-    var data = List<T>.of(_originalData);
+    late List<T> data;
+    if (sortOriginalData) {
+      data = _modifiableOriginalData;
+    } else {
+      // TODO(kenz): copying the list for every sort could cause performance
```

Comment:
Did you add this option because of performance issues in this method? If so, I don't see an issue with setting `sortOriginalData` as the default since the data is typically only being fed to the table controller and is replaced / added to rather than being used for additional processing, so order shouldn't really matter.

Reply:
In some places, the original list could be the same list being used by the controller, so I decided to make this opt-in for now to ensure we don't break anything in other views. I agree it would be beneficial to make this behavior the default instead of the copy method that we have been using, but I wanted to do an audit of this behavior outside of the scope of this PR.

Reply:
Added a comment to https://github.com/flutter/devtools/issues/5091